src/ff1.rs

```diff
@@ -388,6 +388,36 @@ mod tests {
                     33, 28, 8, 10, 0, 10, 35, 17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13,
                 ],
             },
+            // From https://github.com/capitalone/fpe/blob/master/ff1/ff1_test.go
+            TestVector {
+                aes: AesType::AES256,
+                key: vec![
+                    0x2B, 0x7E, 0x15, 0x16, 0x28, 0xAE, 0xD2, 0xA6, 0xAB, 0xF7, 0x15, 0x88, 0x09,
+                    0xCF, 0x4F, 0x3C, 0xEF, 0x43, 0x59, 0xD8, 0xD5, 0x80, 0xAA, 0x4F, 0x7F, 0x03,
+                    0x6D, 0x6F, 0x04, 0xFC, 0x6A, 0x94,
+                ],
+                radix: 36,
+                tweak: vec![],
+                pt: vec![
+                    33, 28, 8, 10, 0, 10, 35, 17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13, 33,
+                    28, 8, 10, 0, 10, 35, 17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13, 33, 28, 8,
+                    10, 0, 10, 35, 17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13, 33, 28, 8, 10, 0,
+                    10, 35, 17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13, 33, 28, 8, 10, 0, 10,
+                    35, 17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13, 33, 28, 8, 10, 0, 10, 35,
+                    17, 2, 10, 31, 34, 10, 21, 34, 35, 30, 32, 13, 33, 28, 8, 10, 0, 10, 35, 17, 2,
+                    10, 31, 34, 10, 21,
+                ],
+                // lwulibfp1ju3ksztumqomwenpv7duy9q7pg7zf3eg3rjlfy46gmgkqjfwvromfjjktmbey8meqk9zkcmgvkv4s9ll5ctozme1hf15w7xo6zsylqcr0nbx9jbf10umzok
+                ct: vec![
+                    21, 32, 30, 21, 18, 11, 15, 25, 1, 19, 30, 3, 20, 28, 35, 29, 30, 22, 26, 24,
+                    22, 32, 14, 23, 25, 31, 7, 13, 30, 34, 9, 26, 7, 25, 16, 7, 35, 15, 3, 14, 16,
+                    3, 27, 19, 21, 15, 34, 4, 6, 16, 22, 16, 20, 26, 19, 15, 32, 31, 27, 24, 22,
+                    15, 19, 19, 20, 29, 22, 11, 14, 34, 8, 22, 14, 26, 20, 9, 35, 20, 12, 22, 16,
+                    31, 20, 31, 4, 28, 9, 21, 21, 5, 12, 29, 24, 35, 22, 14, 1, 17, 15, 1, 5, 32,
+                    7, 33, 24, 6, 35, 28, 34, 21, 26, 12, 27, 0, 23, 11, 33, 9, 19, 11, 15, 1, 0,
+                    30, 22, 35, 24, 20,
+                ],
+            },
```

Comment:
I checked these test vectors against the implementation from https://git.code.sf.net/p/format-preserving-encryption/code . For completeness, the Java program used was:
```
package FFOne;

import javax.xml.bind.DatatypeConverter;
import javax.crypto.spec.SecretKeySpec;
import java.security.InvalidKeyException;
import java.util.Arrays;
import org.fpe4j.FF1;

public class FFOne {
    public static void main(String[] args) {
        try {
            byte[] tweak = {};
            SecretKeySpec key = new SecretKeySpec(DatatypeConverter.parseHexBinary(
                "2B7E151628AED2A6ABF7158809CF4F3CEF4359D8D580AA4F7F036D6F04FC6A94"), "AES");
            FF1 ff1 = new FF1(36, 0);
            String plainstr = "xs8a0azh2avyalyzuwdxs8a0azh2avyalyzuwdxs8a0azh2avyalyzuwdxs8a0azh2avyalyzuwdxs8a0azh2avyalyzuwdxs8a0azh2avyalyzuwdxs8a0azh2avyal";
            int[] plaintext = decode(plainstr);
            assert encode(plaintext).equals(plainstr);
            int[] ciphertext = ff1.encrypt(key, tweak, plaintext);
            int[] decrypted  = ff1.decrypt(key, tweak, ciphertext);
            System.out.println(Arrays.toString(plaintext));
            System.out.println(encode(ciphertext));
            System.out.println(Arrays.toString(ciphertext));
            System.out.println(encode(decrypted));
            assert encode(decrypted) == plainstr;
        } catch (InvalidKeyException e) {
            throw new RuntimeException(e);
        }
    }

    public static int[] decode(String s) {
        return s.chars().map(c -> {
            if (c >= '0' && c <= '9') return c-'0';
            if (c >= 'a' && c <= 'z') return c-'a'+10;
            throw new IllegalArgumentException();
        }).toArray();
    }
    public static String encode(int[] a) {
        StringBuilder sb = new StringBuilder(a.length);
        Arrays.stream(a).map(i -> {
            if (i >=  0 && i <=  9) return '0'+i;
            if (i >= 10 && i <= 35) return 'a'+(i-10);
            throw new IllegalArgumentException();
        }).forEach(sb::appendCodePoint);
        return sb.toString();
    }
}
```